checks/linux.py

```diff
@@ -0,0 +1,172 @@
+    isDistroNix = search('Distribution:', lines)
+    isFlatpak = search('Flatpak Runtime:', lines)
+
+    if (len(isDistroNix) <= 0) and (len(isFlatpak) <= 0):
+        return
+
+    sessionTypeLine = getSessionTypeLine(lines)
+    if not sessionTypeLine:
+        return
+
+    if 'wayland' not in sessionTypeLine:
+        return
+
+    if len(isDistroNix) > 0:
+        if '"Ubuntu" "20.04"' in isDistroNix[0]:
+            return [LEVEL_CRITICAL, "Ubuntu 20.04 under Wayland",
+                    "Ubuntu 20.04 does not provide the needed dependencies for OBS to capture under Wayland.<br> So OBS is able to capture only under X11/Xorg."]
+
+    windowSystemLine = getWindowSystemLine(lines)
+    # If there is no Window System, OBS is running under Wayland
+    if windowSystemLine:
+        # If there is, OBS is running under XWayland
+        return [LEVEL_CRITICAL, "Running under XWayland",
+                "OBS is running under XWayland, which prevents OBS from being able to capture.<br>To fix that, you will need to run OBS with the following command in a terminal:<p><code>obs -platform wayland</code></p>"]
+
+    hasNoPipewireCapture = search('[pipewire] No capture', lines)
+    if len(hasNoPipewireCapture) > 0:
+        return [LEVEL_CRITICAL, "No PipeWire capture on Wayland",
+                """In order to capture displays or windows under Wayland, OBS requires the appropriate PipeWire capture portal for your Desktop Environment.<br><br>
+                An overview of available capture portals can be found on the Arch Linux Wiki:<br>
+                <a href='https://wiki.archlinux.org/title/XDG_Desktop_Portal'>XDG Desktop Portal</a><br>
+                Note that the availability of Window and/or Display capture depends on your Desktop Environment's implementation of these portals."""]
+
+    return [LEVEL_INFO, "Wayland", ""]
+
+
+def checkX11Captures(lines):
+    isDistroNix = search('Distribution:', lines)
+    isFlatpak = search('Flatpak Runtime:', lines)
+
+    if (len(isDistroNix) <= 0) and (len(isFlatpak) <= 0):
+        return
+
+    sessionTypeLine = getSessionTypeLine(lines)
+    if not sessionTypeLine:
+        return
+
+    if 'x11' not in sessionTypeLine:
+        return
+
+    # obsolete PW sources
+    hasPipewireCaptureDesktop = search('pipewire-desktop-capture-source', lines)
+    hasPipewireCaptureWindow = search('pipewire-window-capture-source', lines)
+    # unified PW source
+    hasPipewireCaptureScreen = search('pipewire-screen-capture-source', lines)
+
+    if (len(hasPipewireCaptureDesktop) > 0) or (len(hasPipewireCaptureWindow) > 0) or (len(hasPipewireCaptureScreen) > 0):
+        return [LEVEL_WARNING, "PipeWire capture on X11",
+                """While technically possible, most Desktop Environments do not implement the PipeWire capture portals on X11. Your captures will therefore likely not work, i.e. you will be unable to pick a window or display, or the selected source will stay empty.<br><br>
```

Comment:
```suggestion
                """Most Desktop Environments do not implement the PipeWire capture portals on X11. This can result in being unable to pick a window or display, or the selected source will stay empty.<br><br>
```

Reply:
Done.